### ISSUE TITLE
brightness matebook 13 d'ont work on Monterey

### ISSUE THREAD

**Issue by TheKiller57** [ske1996/matebook-13and14-OpenCore-Hackintosh#187]
Hello guys
I have monterey installed on my MateBook 13 8go i7 8th, but brightness isn't working..
Can someone help me ?

**Comment by chewzzqq** [comment]
Hi TheKiller57, did you follow Ske1996's 开启HIDPI procedure?
You need follow step by step by Ske1996.

Myself using Huawei Matebook 13 i5 8th, follow exactly the whole procedure by Ske1996 without any problem.
From bios 1.28 downgrade to 1.27 solve the issue wont wakeup after sleep mode.
Also triple boot , Monterey, Windows 10 and Mx Linux.
Brightness working, wifi working, sleep mode working, sound working, working usage can last 4 hour.

Awesome Ske1996 skill and procedure, almost full working and stable , thanks Ske1996.

**Comment by TheKiller57** [comment]
Hello
Thank you for your answer, the only step that I did not follow is the bios downgrade, because I have a bin file but I do not know how to flash it ..

**Comment by chewzzqq** [comment]
Hi TheKiller57, 

You can check on here.
[https://www.reddit.com/r/MatebookXPro/comments/i7ep5d/guide_downgrading_bios_for_huawei_matebook_13_2019/](https://www.reddit.com/r/MatebookXPro/comments/i7ep5d/guide_downgrading_bios_for_huawei_matebook_13_2019/)

And also disable the SIP in recovery.
Hope it can help you.

**Comment by TheKiller57** [comment]
The link of the soft is broken, private..

**Comment by ske1996** [comment]
check if you activate FN before you press F1 or F2

**Comment by TheKiller57** [comment]
It's activated :(

**Comment by ske1996** [comment]
**Device info（设备信息）**

Matebook xx 20xx  

CPU:  
RAM:  
SSD model:
MacOS version:   
EFI release version:

**Describe the bug（问题简述）**
A clear and concise description of what the bug is.

**To Reproduce（详述问题发生前的操作）**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error


**Screenshots（截图）**
If applicable, add screenshots to help explain your problem.

**Additional context（其他附件事项）**
Add any other context about the problem here.

**Comment by ske1996** [comment]
and upload a video that shows your brightness key doesnt work.

**Comment by TheKiller57** [comment]
Matebook 13 2019
i5 8th 8go 256Go
macOS Monterey
EFI with all kext and oc file updated
BIOS 1.30
The task bar for brightness not working, I can't set brightness level in setting, in task bar, or with keyboard.

**Comment by ske1996** [comment]
> Matebook 13 2019 i5 8th 8go 256Go macOS Monterey EFI with all kext and oc file updated BIOS 1.30 The task bar for brightness not working, I can't set brightness level in setting, in task bar, or with keyboard.

i trust what you said,and i did help hundreds people to build their hackintosh,and none has brightness isuess.

so,if you can show me a video thatshows your brightness key doesnt work, and shows your keyborad and your screen clearly.

it will be very helpful

**Comment by TheKiller57** [comment]
https://www57.zippyshare.com/v/KdKxiVNk/file.html
No problem this is the video link

**Comment by ske1996** [comment]
maybe i am not 100% right,but i think this isuess probably related to bios version.

i've heard someone upgrade his bios to 1.30 and caused a not-wake isuess.

**Comment by TheKiller57** [comment]
Can I downgrade to 1.27 bios ?

**Comment by ske1996** [comment]
> https://www57.zippyshare.com/v/KdKxiVNk/file.html No problem this is the video link

i've never uses that site before and dont know how to download in that site,

cloud you upload your video here?

drop your video file here with your reply,it will be allowed

**Comment by TheKiller57** [comment]
The video is to big :p

**Comment by ske1996** [comment]
> Can I downgrade to 1.27 bios ?

i 've  not trid that by myself before,but a guy in chat group told me it worked.

**Comment by TheKiller57** [comment]
https://drive.google.com/file/d/1wpCPBkyW6zJeONRa_SZJTMGy44CoAagM/view?usp=sharing

**Comment by ske1996** [comment]
> https://drive.google.com/file/d/1wpCPBkyW6zJeONRa_SZJTMGy44CoAagM/view?usp=sharing

TY and i checked.

**Comment by TheKiller57** [comment]
And how to downgrade ? I have the .bin bios file but no software for update it

**Comment by ske1996** [comment]
> https://drive.google.com/file/d/1wpCPBkyW6zJeONRa_SZJTMGy44CoAagM/view?usp=sharing

cloud you upload your EFI folder here?

**Comment by ske1996** [comment]
> And how to downgrade ? I have the .bin bios file but no software for update it

i dont know how to do,but i know it cloud be downloaded

**Comment by TheKiller57** [comment]
Yes 1 minute

**Comment by TheKiller57** [comment]
https://drive.google.com/file/d/14y1zFa1uBww063EV6o7WmXfaJ8D16tld/view?usp=sharing

**Comment by ske1996** [comment]
seems like you add some ssdt which dont from me and deleted disable dgpu properties.

have you trid my original files？

was it same in result？

**Comment by TheKiller57** [comment]
Yes, I tried all the EFIs and it was the only one that allowed me to start and keep iMessages

**Comment by ske1996** [comment]
> Yes, I tried all the EFIs and it was the only one that allowed me to start and keep iMessages

……mine need to edit ssn and platform infos before you use.
or it wont boot

i wrote that in readme.

**Comment by TheKiller57** [comment]
I don't know where this EFI comes from, by dint of looking for and testing all the efi lol

**Comment by chewzzqq** [comment]
Hi TheKiller57,

I have try to downgrade from my Huawei Matebook 13 (2018-2019) from 1.28 to 1.27 .
A Flash BIOS upgrade/downgrade runs the risk of making a board unbootable. 
Do it at your own risk.

- changed flag to 0 for [Bios_Version_Check] in platform.ini;

- unpacked 1.27 bios file UEFI_FW.bin (from Ske1996) in another folder ("temp"), and renamed to isflash.bin;

- replaced isflash.bin in "Huawei Matebook 13 Bios Downgrade" folder with 1.27 bin from "temp";

- started H2OFFT-Wx64.exe with AC connected, followed advices, rebooted.

**Comment by TheKiller57** [comment]
Ok succefully downfgrade, i tried to unlock cfg and set dvmt

**Comment by ske1996** [comment]
> Hi TheKiller57,
> 
> You must use Ske1996's EFI for Monterey. Follow 100% procedure and files provided by Ske1996.
> 
> I had successfully downgrade from my Huawei Matebook 13 (2018-2019) from 1.28 to 1.27 . A Flash BIOS upgrade/downgrade runs the risk of making a board unbootable. Please do it at your own risk.
> 
> [Download from here](https://send.cm/d/74qK)
> 
> * changed flag to 0 for [Bios_Version_Check] in platform.ini;
> * unpacked 1.27 bios file UEFI_FW.bin (from Ske1996) in another folder ("temp"), and renamed to isflash.bin;
> * replaced isflash.bin in "Huawei Matebook 13 Bios Downgrade" folder with 1.27 bin from "temp";
> * started H2OFFT-Wx64.exe with AC connected, followed advices, rebooted (Once reboot, the downgrade will immediately take effect) .
> 
> After successfully downgrade bios, please unlock again CFG and modified dvmt. And also disable the SIP as well in recovery.

Thank you for supporting me,
btw,disabling SIP is not necessary from macOS 11.1,and i recommend to keep it enabled. For security reason.
Of course,that probability of your laptop under attacking close to zero,lol.

**Comment by chewzzqq** [comment]
Dear Ske1996,

Noted with thanks.
